scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/CascadingBackend.scala

```diff
@@ -422,7 +422,11 @@ object CascadingBackend {
       val cpipe = toPipeUnoptimized[A](optPipe, dest.sinkFields)(fd, mode, dest.setter)
       dest.writeFrom(cpipe)(fd, mode)
     }
-    todos.foreach(doWrite(_))
+    if (todos.nonEmpty) {
+      todos.foreach(doWrite(_))
+      // In case if during writes planning other typed writes happened.
+      planTypedWrites(fd, mode)
```

Comment:
is `planTypedWrites` tailrec?

Reply:
I'm not particularly nitty on this but overall I prefer do not annotate methods with tailrec unless they require it.

Reply:
I think it is tailrec, but I do also worry about an infinite loop here with a buggy implementation of writeFrom.

Maybe we can do an info log if we see this and recommend users not do new writes in writes. We could have an inner method that keeps a count of how deep we go in the recursion, and if depth > 0 we do the info log message?

Reply:
I did log warn message if you need to apply `planTypedWrites` twice and exception if you need to apply more.